File: e2e/testdata/lifecycle/python/stop.py
import os
import urllib.request


def new():
    return Function()


class Function:
    def __init__(self):
        self._recorder_url = os.environ.get("RECORDER_URL", "")

    def stop(self):
        if self._recorder_url:
            req = urllib.request.Request(
                self._recorder_url + "/record?id=stop-python",
                method="POST",
            )
            try:
                urllib.request.urlopen(req, timeout=5)
            except Exception:
                pass

    async def handle(self, scope, receive, send):
        await send({
            "type": "http.response.start",
            "status": 200,
            "headers": [[b"content-type", b"text/plain"]],
        })
        await send({
            "type": "http.response.body",
            "body": b"OK",
        })

File: e2e/testdata/lifecycle/go/stop.go
package function

import (
	"context"
	"fmt"
	"net/http"
	"os"
)

type Function struct {
	recorderURL string
}

func New() *Function {
	return &Function{recorderURL: os.Getenv("RECORDER_URL")}
}

func (f *Function) Stop(_ context.Context) error {
	if f.recorderURL != "" {
		resp, err := http.Post(f.recorderURL+"/record?id=stop-go", "text/plain", nil)
		if err == nil {
			resp.Body.Close()
		}
	}
	return nil
}

func (f *Function) Handle(w http.ResponseWriter, _ *http.Request) {
	fmt.Fprint(w, "OK")
}
